src/commands.rs

```diff
@@ -461,6 +461,41 @@ pub fn list(ctx: &mut Context<'_>, no_connect: bool) -> anyhow::Result<()> {
   Ok(())
 }
 
+/// Fill the SD card with random data
+pub fn fill(ctx: &mut Context<'_>) -> anyhow::Result<()> {
+  with_storage_device(ctx, |ctx, mut device| {
+    let pin_entry = pinentry::PinEntry::from(args::PinType::Admin, &device)?;
+
+    // Similar to reset, we want the user to re-enter the admin PIN even if is cached to avoid
+    // accidental data loss.
+    pinentry::clear(&pin_entry).context("Failed to clear cached secret")?;
```

Comment:
Fine to leave it for now because there is precedent, but couldn't we use the "no caching" feature here to force reentry without having to clear it for all other commands? Seems as if `PinEntry` could just be extended with a flag. Or perhaps that has not the desired effect and it just causes the agent itself to not create a new entry but still take any existing ones.

Reply:
I think it should work because we don’t set the cache ID if `no_cache` is true, so `pinentry` should not have a way to use the cached value.  I’ll have a look at it, but I’d rather do that in a separate PR.

Reply:
After giving this issue some more thought:  If we would use `no_cache`, this would also mean that the user has to re-enter the admin PIN for the next command.  This is not really what we want.

Reply:
Ah yeah. Good point. Let's leave it as-is.